spec/models/validators/valid_date_presence_validator_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe ValidDatePresenceValidator do
+  let(:test_model) do
+    Class.new do
+      include ActiveModel::Validations
+
+      attr_accessor :date_attr, :other_attr
+
+      validates :date_attr, valid_date_presence: true
+    end
+  end
+
+  subject { test_model.new }
+
+  it 'passes when the date is valid' do
+    expect_any_instance_of(ValidDatePresenceValidator).to receive(:valid_date?).and_return true
```

Comment:
I think I should add a new test in each validator spec to make sure it is checking for an invalid date too (for `DateBeyondOsraEstablishmentValidator` and `DateNotInFutureValidator`). 
At the moment I am only doing it in `ValidDatePresenceValidator` (in `valid_date_presence_validator_spec.rb` spec) which checks for valid and invalid dates .

Like:
```ruby
  it "allows date attribute to be an invalid date format" do
    expect_any_instance_of(DateBeyondOsraEstablishmentValidator).to receive(:valid_date?).and_return false
    expect(subject).to allow_value("invalid_date_format").for :date_attr
  end
#same for: DateNotInFutureValidator
```

In the model,  to block invalid dates, `validates :start_date, valid_date_presence: true` is needed.